### DISCUSSION TITLE
Trouble bridging to a TLS-enabled broker

### DISCUSSION THREAD

**Discussion by tjcruz-dei** [emqx/emqx#7507] [Q&A]
Dear All,

I tried bridging with a TLS-enabled MQTT broker following the rule-based approach, with no luck. I found this message in the logs:

"2022-04-02T17:35:21.259935+00:00 [error] [Bridge Connect] Failed to connect with module=emqx_bridge_mqtt, config=[{username,"xxxxxxx"},{subscriptions,[]},{start_type,auto},{ssl_opts,[{verify,verify_none},{versions,['tlsv1.3','tlsv1.2','tlsv1.1',tlsv1]},{ciphers,["TLS_AES_256_GCM_SHA384","TLS_AES_128_GCM_SHA256","TLS_CHACHA20_POLY1305_SHA256","TLS_AES_128_CCM_SHA256","TLS_AES_128_CCM_8_SHA256","ECDHE-RSA-AES128-GCM-SHA256","ECDHE-ECDSA-AES256-GCM-SHA384","ECDHE-RSA-AES256-GCM-SHA384","ECDHE-ECDSA-AES256-SHA384","ECDHE-RSA-AES256-SHA384","ECDHE-ECDSA-DES-CBC3-SHA","ECDH-ECDSA-AES256-GCM-SHA384","ECDH-RSA-AES256-GCM-SHA384","ECDH-ECDSA-AES256-SHA384","ECDH-RSA-AES256-SHA384","DHE-DSS-AES256-GCM-SHA384","DHE-DSS-AES256-SHA256","AES256-GCM-SHA384","AES256-SHA256","ECDHE-ECDSA-AES128-GCM-SHA256","ECDHE-ECDSA-AES128-SHA256","ECDHE-RSA-AES128-SHA256","ECDH-ECDSA-AES128-GCM-SHA256","ECDH-RSA-AES128-GCM-SHA256","ECDH-ECDSA-AES128-SHA256","ECDH-RSA-AES128-SHA256","DHE-DSS-AES128-GCM-SHA256","DHE-DSS-AES128-SHA256","AES128-GCM-SHA256","AES128-SHA256","ECDHE-ECDSA-AES256-SHA","ECDHE-RSA-AES256-SHA","DHE-DSS-AES256-SHA","ECDH-ECDSA-AES256-SHA","ECDH-RSA-AES256-SHA","AES256-SHA","ECDHE-ECDSA-AES128-SHA","ECDHE-RSA-AES128-SHA","DHE-DSS-AES128-SHA","ECDH-ECDSA-AES128-SHA","ECDH-RSA-AES128-SHA","AES128-SHA"]}]},{ssl,true},{retry_interval,20},{proto_ver,v3},{password,'xxxxxxxx'},{max_inflight_batches,32},{keepalive,60},{if_record_metrics,false},{forward_mountpoint,[]},{disk_cache,false},{clientid,"clientUC_0005DBAF51C19FE0F440000018FD0000"},{clean_start,true},{bridge_mode,true},{address,"xxxxxxxxxxxxx.hivemq.cloud:8883"}], reason:{unauthorized_client,undefined}"

**My configuration:**
Remote broker: xxxxxxxxxxxxx.hivemq.cloud:8883
SSL enabled
Bridge mode: true
CA certificates: /etc/emqx/certs/cacert.pem
SSL Certfile: /etc/emqx/certs/client-cert.pem
SSL Keyfile: /etc/emqx/certs/client-key.pem
SSL Cyphers: ECDHE-RSA-AES128-GCM-SHA256, ECDHE-ECDSA-AES256-GCM-SHA384,ECDHE-RSA-AES256-GCM-SHA384,ECDHE-ECDSA-AES256-SHA384,ECDHE-RSA-AES256-SHA384,ECDHE-ECDSA-DES-CBC3-SHA,ECDH-ECDSA-AES256-GCM-SHA384,ECDH-RSA-AES256-GCM-SHA384,ECDH-ECDSA-AES256-SHA384,ECDH-RSA-AES256-SHA384,DHE-DSS-AES256-GCM-SHA384,DHE-DSS-AES256-SHA256,AES256-GCM-SHA384,AES256-SHA256,ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-RSA-AES128-GCM-SHA256,ECDHE-ECDSA-AES128-SHA256,ECDHE-RSA-AES128-SHA256,ECDH-ECDSA-AES128-GCM-SHA256,ECDH-RSA-AES128-GCM-SHA256,ECDH-ECDSA-AES128-SHA256,ECDH-RSA-AES128-SHA256,DHE-DSS-AES128-GCM-SHA256,DHE-DSS-AES128-SHA256,AES128-GCM-SHA256,AES128-SHA256,ECDHE-ECDSA-AES256-SHA,ECDHE-RSA-AES256-SHA,DHE-DSS-AES256-SHA,ECDH-ECDSA-AES256-SHA,ECDH-RSA-AES256-SHA,AES256-SHA,ECDHE-ECDSA-AES128-SHA,ECDHE-RSA-AES128-SHA,DHE-DSS-AES128-SHA,ECDH-ECDSA-AES128-SHA,ECDH-RSA-AES128-SHA,AES128-SHA

I'm really out of ideas.

Thanks, in advance, for any hints or help.
--TC

**Comment by zmstone** [comment]
duplicates: https://github.com/emqx/emqx/issues/7506
